documentation/src/main/asciidoc/public/reference/_search-dsl-aggregation.adoc

```diff
@@ -322,6 +322,27 @@ See <<search-dsl-argument-type>> for more information.
 * For fields in nested objects, all nested objects are considered by default,
 but that can be <<search-dsl-aggregation-common-filter,controlled explicitly with `.filter(...)`>>.
 
+[[search-dsl-aggregation-withparameters]]
+== `withParameters`: create aggregations using query parameters
+
+include::../components/_incubating-warning.adoc[]
+
+The `withParameters` aggregation allows building aggregations using <<search-dsl-query-parameters,query parameters>>.
+
+This type of aggregation gives user a context exposing an aggregation factory and access to query parameters and delays creating
+an actual aggregation to the query building time allowing all query parameters to be set.
```

Comment:
I would avoid the word "user" since it's unclear whether we talk about developers or end users.

```suggestion
This type of aggregation requires a function that accepts query parameters and returns an aggregation. That function will get called at query building time.
```

Same for other documentation.